woocommerce/woocommerce-android
# Issue/2406 read only product type

Fixes #2406 and #2297 by adding an extra row in Product detail screen to display the product type.

As mentioned in #2406. the mapping between the product types are:
- Physical product (`type` is `simple`, `virtual` field is off)
- Virtual product (`type` is `simple`, `virtual` field is on)
- Variable product (`type` is `variable`)
- Grouped product (`type` is  `grouped`)
- External product (`type` is `external`)

This PR also removes the `isAddEditProductRelease1Enabled` method from the code and opens up the editing process for all the product types, except variations (which is being worked on by @0nko separately).

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/86319844-0a56f580-bc53-11ea-8583-a8b62674e845.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/86319850-0c20b900-bc53-11ea-9aee-96a5144eb623.png" width="300"/> 

#### Testing
- Click on a physical product from the Products tabs and notice that the product type is displayed correctly.
- Click on a virtual product from the Products tabs and notice that the product type is displayed correctly.
- Click on an external product from the Products tabs and notice that the product type is displayed correctly.
- Click on a grouped product from the Products tabs and notice that the product type is displayed correctly.

~**Note that this PR is in draft till [this PR](https://github.com/woocommerce/woocommerce-android/pull/2602) can be merged.**~

Update release notes:
